src/statemachine/blender.js

```diff
@@ -0,0 +1,244 @@
+    }
+
+    function smooth(start, end, t) {
+        return interpolate.value(start, end, pc.math.smootherstep(0, 1, t));
+    }
+
+    /**
+     * @callback
+     * @name pc.interpolate~valuecb
+     * @description call back containing the value of an interpolation
+     * @param {pc.Vec3|pc.Quat|Number} value The current value
+     * @param {Number} t The current proportion of the interpolation that is complete 0..1
+     */
+
+    var interpolate = {
+        /**
+         * @name pc.interpolate.value
+         * @function
+         * @description Get a value between start and end with unconstrained t (can be > 1 or < 0)
+         * @param {pc.Vec3|pc.Quat|Number} start The start point for the interpolation
+         * @param {pc.Vec3|pc.Quat|Number}end The end point for the interpolation
+         * @param {Number} t The position between start and end
+         * @returns {pc.Vec3|pc.Quat|Number} The point at the specified time
+         */
+        value: value,
+        /**
+         * @name pc.interpolate.lerp
+         * @function
+         * @description lerp a value between start and end with constrained t (forced between 0 and
+         *     1)
+         * @param {pc.Vec3|pc.Quat|Number} start The start point for the interpolation
+         * @param {pc.Vec3|pc.Quat|Number} end The end point for the interpolation
+         * @param {Number} t The position between start and end
+         * @returns {pc.Vec3|pc.Quat|Number} The lerped value
+         */
+        lerp: lerp,
+        /**
+         * @name pc.interpolate.smooth
+         * @function
+         * @description Use smoothstep to make a natural lerp of a value between start and end with
+         *     constrained t (forced between 0 and 1)
+         * @param {pc.Vec3|pc.Quat|Number} start The start point for the interpolation
+         * @param {pc.Vec3|pc.Quat|Number} end The end point for the interpolation
+         * @param {Number} t The position between start and end
+         * @returns {pc.Vec3|pc.Quat|Number} The smoothly interpolated value
+         */
+        smooth: smooth,
+        /**
+         * @name pc.interpolate.moveTowards
+         * @function
+         * @description Move a value towards a target with a limited maximum step
+         * @param {pc.Vec3|pc.Quat|Number} start The start point for the move
+         * @param {pc.Vec3|pc.Quat|Number}end The target for the move
+         * @param {Number} max The maximum distance to move, if start and end are vectors or
+         *     numbers then this is a distance, if start and are quaternions then this is a
+         *     distance measured in radians
+         * @example
+         * update: function(dt) {
+		 *     this.entity.setPosition(pc.interpolate.moveTowards(this.entity.getPosition(),
+		 *     someTargetPosition, this.speed * dt));
+		 * }
+         * @returns {pc.Vec3|pc.Quat|Number} The updated value
+         */
+        moveTowards: function (start, end, max) {
```

Comment:
Yeah good point. Will do. Now I know how to run the engine locally I'll make up a demo pack

Sent from my iPhone

> On 19 Jun 2015, at 10:28, vkalpias <notifications@github.com> wrote:
> 
> In src/statemachine/blender.js:
> 
> > +         * @name pc.interpolate.moveTowards
> > +         * @function
> > +         * @description Move a value towards a target with a limited maximum step
> > +         * @param {pc.Vec3|pc.Quat|Number} start The start point for the move
> > +         * @param {pc.Vec3|pc.Quat|Number}end The target for the move
> > +         * @param {Number} max The maximum distance to move, if start and end are vectors or
> > +         *     numbers then this is a distance, if start and are quaternions then this is a
> > +         *     distance measured in radians
> > +         * @example
> > +         * update: function(dt) {
> > +		 *     this.entity.setPosition(pc.interpolate.moveTowards(this.entity.getPosition(),
> > +		 *     someTargetPosition, this.speed * dt));
> > +		 * }
> > +         * @returns {pc.Vec3|pc.Quat|Number} The updated value
> > +         */
> > +        moveTowards: function (start, end, max) {
> So I noticed this method and probably others too are making clones of vectors, quats etc. Since these are methods that are meant to be called every frame do you think maybe you could use scratch vectors/quats for temporary operations instead so that we minimize the memory footprint?
> 
> —
> Reply to this email directly or view it on GitHub.
> 

-- 
See our 2 minute video on DBS bringing gamification to Singapore here: 
www.3radical.com/resources/videos/#DBSVideo


3radical Limited, 1 Friary, Temple Quay, Bristol BS1 6EA, England.  
T: +44 117 344 5008 ¦ F : +44 117 344 5188

3radical Pte Ltd, 3 Church Street, Level 8 Samsung Hub, Singapore 049483.  
T: +65 6408 3711 ¦ F : +65 6408 0101

www.3radical.com

3radical Limited is a limited company registered in England and Wales 
under number 7872556.  The registered address of 3radical Limited is 1 
Friary, Temple Quay, Bristol BS1 6EA, England.

3radical Pte Ltd is a registered in Singapore under number 201312239K.  The 
registered address of 3radical Pte Ltd is 51 Goldhill Plaza, #07-10/11, 
Singapore 308900.

The information in this e-mail and any attachments may be confidential.  If 
you are not the intended recipient, please notify the sender as soon as 
possible and delete any copies. Unauthorised use, dissemination, 
distribution, publication or copying of this communication is prohibited.

Attachments to this email have been checked with virus detection 
software before transmission, however you should carry out virus checks 
before opening any attachment. 3radical Limited and 3radical Pte Ltd 
accepts no liability for any loss or damage which may be caused by software 
viruses.